python/graphstorm/gconstruct/construct_payload_graph.py

```diff
@@ -0,0 +1,592 @@
+
+    {
+        "edge_type": "<edge type>",
+        "src_node_id": <src edge id>,
+        "dest_node_id": <dest edge id>
+        "features: {
+            "<feat_name>": [feat_val]
+        }
+    }
+
+    Parameters:
+    ------------
+        gconstruct_edge_conf_list: dict
+            GConstruct node configuration.
+        payload_edge_conf_list: dict
+            Payload Edge configuration.
+        node_id_map: dict
+            A mapping from original node string ID to unique integer IDs.
+    Returns:
+    -------
+        edges: {etype: (src_np_array, dst_np_array)}
+            Edges information with mapped integer IDs.
+        edge_data: {efeat_name: edge_feat_np_array}
+            Edge features with transformed data.
+    """
+    edges = {}
+    edge_data = {}
+    merged_payload_edge_conf_list = merge_payload_input(payload_edge_conf_list)
+    for edge_conf in merged_payload_edge_conf_list:
+        edge_type = edge_conf["edge_type"]
+        src_node_ids = np.array(edge_conf["src_node_id"])
+        dest_node_ids = np.array(edge_conf["dest_node_id"])
+        gconstruct_edge_conf = get_conf(gconstruct_edge_conf_list, edge_type, "Edge")
+
+        # List is not hashable
+        edge_type = tuple(edge_type)
+        (feat_ops, two_phase_feat_ops, after_merge_feat_ops, hard_edge_neg_ops) = \
+            parse_feat_ops(gconstruct_edge_conf["features"],
+                           gconstruct_edge_conf["format"]["name"]) \
+                if 'features' in gconstruct_edge_conf else (None, [], {}, [])
+
+        id_map = {edge_type[0]: node_id_map[edge_type[0]],
+                  edge_type[2]: node_id_map[edge_type[2]]}
+
+        # For edge hard negative transformation ops, more information is needed
+        for op in hard_edge_neg_ops:
+            op.set_target_etype(edge_type)
+            op.set_id_maps(id_map)
+
+        # Always do single process feature transformation as there is only payload input
+        if feat_ops is not None:
+            if "features" not in edge_conf:
+                raise MissingValError("features", "edge payload")
+            input_feat = edge_conf["features"]
+            # Input features raw data should be numpy array type
+            for key, val in input_feat.items():
+                input_feat[key] = np.array(val)
+            if len(two_phase_feat_ops) > 0:
+                phase_one_ret = prepare_data(input_feat, two_phase_feat_ops)
+                update_two_phase_feat_ops(phase_one_ret, two_phase_feat_ops)
+            feat_data, _ = process_features(input_feat, feat_ops, None)
+        else:
+            feat_data = {}
+        for feat_name in list(feat_data):
+            if feat_name in after_merge_feat_ops:
+                # do data transformation with the entire feat array.
+                merged_feat = \
+                    after_merge_feat_ops[feat_name].after_merge_transform(feat_data[feat_name])
```

Comment:
Maybe avoid using `\` and follow PEP8?
```suggestion
                merged_feat = (
                    after_merge_feat_ops[feat_name]
                        .after_merge_transform(feat_data[feat_name])
                )
```